tests/ports/psoc6/README.md

```diff
@@ -0,0 +1,18 @@
+# PSoC6 Port Tests
```

Comment:
In general I am happy with this organization. I just have couple of suggestions which can be taken into consideration/discussion now or later:
1. We have tests with different configurations, some require multiple devices to be passed while others only one etc. I can see this is organized in folder structure now. We should please have the same in test script also. For ex: 
a. Possibility to run easily single tests:
./tests/ports/psoc6/run_psoc6_tests.sh --test-suite hw_ext -single pwm -d ${devs[0]}
b. From ci, it can still be full and internally it calls individual tests like:
./tests/ports/psoc6/run_psoc6_tests.sh --test-suite hw_ext -d ${devs[0]}
 which in turn can run all hw_ext related tests.
I would agree that we can refactor this post v1 release or while thinking to migrate to python.
2. Test execution commands help in README.md for **atleast** test-suites.
3. Suggestion for some namings:
psoc6/
├─ extended_hw_tests/     
│  ├─ diagrams/       
│  ├─ multi_board/  
│  ├─ single_board/
│  ├─ **README** --> We don't need this README or?
├─ on_device_mp_tests/
├─ board_only_tests/ 
│  ├─ multi_board/  
│  ├─ single_board/        
├─ run_psoc6_tests.sh  
├─ README.md
But this is only my suggestions. Feel free to consider or not :)

Reply:
1.a) yep, agree. Easier with the python migration + yaml
2. Ok. I would mention that there is a help and add some examples to the help, this way we keep docs lean and close to the source.
3.  Good points!
I see your proposal is a bit more self-explanatory. Agree to that. The criteria was to keep them a bit shorter.
I would remove "_tests" suffix, that should be clear being under tests/ports/psoc6.
psoc6/
├─ extended_hw_tests/ --> _test suffix
│ ├─ multi_board/ (ok)
│ ├─ single_board/ (ok)
│ ├─ README --> We don't need this README or? --> This includes the pinout diagrams, and some background info about self-contained tests.
├─ on_device_mp_tests/ --> not sure. The other are also on device. mpremote is also not very descriptive I know. 
├─ board_only_tests/ --> board_only (yes), but then the "board" again in the subcategories is a bit redundant...
│ ├─ multi_board/  --> now _board redundant
│ ├─ single_board/ --> now _board redundant
├─ run_psoc6_tests.sh
├─ README.md

By the way, I am missing the "inputs" folder. Okay, what about?
Any other words for "multi" and "single" describing "several boards" vs "one board"? Maybe, I am not fully convinced of those. "multi" is taken from the already named "multi" script, at least there we try to be consistent. 

psoc6/
**├─ board_extended_hw**  
│ ├─ diagrams/ 
│ ├─ multi/
│ ├─ single/ 
│ ├─ README
**├─ ????? --> Nothing feels accurately descriptive. The only think I can say about this grouping is that we are using mpremote to validate certain functionality, and we are not using a tests suite tool, but custom python scripts talking to the device using mpremote. Some ideas: mp_custom, mp_based, mp_other, mp_non_std.. custom_adv (custom_advanced) or mp_advanced??? What else? can we quick brainstorm?**
**├─ board_only_hw/ --> with the _hw we remove the "only" as "one" nuance?** 
│ ├─ multi/
│ ├─ single/
├─ inputs/
├─ run_psoc6_tests.sh
├─ README.md

Reply:
> Any other words for "multi" and "single" describing "several boards" vs "one board"? Maybe, I am not fully convinced of those. "multi" is taken from the already named "multi" script, at least there we try to be consistent.

_--> I agree. Some more suggestions : 
a) inter --> needs multiple boards and intra --> needs single board
b) distributed and self_contained
c) SBT and MBT (with explanations in README, abbreviations can be clearer?)_

> psoc6/
> **├─ board_extended_hw**
> │ ├─ diagrams/
> │ ├─ multi/
> │ ├─ single/
> │ ├─ README
> **├─ ????? --> Nothing feels accurately descriptive. The only think I can say about this grouping is that we are using mpremote to validate certain functionality, and we are not using a tests suite tool, but custom python scripts talking to the device using mpremote. Some ideas: mp_custom, mp_based, mp_other, mp_non_std.. custom_adv (custom_advanced) or mp_advanced??? What else? can we quick brainstorm?**
> **├─ board_only_hw/ --> with the _hw we remove the "only" as "one" nuance?**
> │ ├─ multi/
> │ ├─ single/
> ├─ inputs/
> ├─ run_psoc6_tests.sh
> ├─ README.md

_Yes the mpremote using tests are harder to name. Nothing really justifies the tests in there. I had initially thought as mp_dev_storage, if we can take out wdt script from there and this can be only for testing storage related functionality. Additionally we then don't need test_inputs folder. It can be inside this tests which also justifies that input files are needed for storage tests. WDT is a bit misfit in that folder and can be added back with other machine tests? We can brainstorm!_

> │ ├─ README --> We don't need this README or? --> This includes the pinout diagrams, and some background info about self-contained tests.

_I would suggest to have **one** README in the "psoc6" level with all information put together under sections. Anyways we are in need to put some explanation for tests there. Also, I don't see the tests to grow further, rather with test plan can be more concise. So maybe like:
psoc6 port is tested for all enabled functionality under following organization:_
- board_extended_hw : 
   a) One line description on multi and single tests
   b) Supported images and description
   c) Commands to run tests in this folder 
- board_only:
   Same as above
- mpremote tests
  Same as above details of needed input files for the test to run.

Reply:
Okay, yes, I can merge the README with the other one.
I wait for the wdt tests to be merge to continue with this PR. 
But I see we are not fully convinced. Let´s just keep "mp_custom" naming. 
Everything will be sufficiently (we´ll try) documented in the README. 

psoc6/
├─ board_extended_hw/
│ ├─ diagrams/
│ ├─ multi/
│ ├─ single/
├─ board_only_hw/
│ ├─ multi/
│ ├─ single/
├─ inputs/
├─ mp_custom/
├─ run_psoc6_tests.sh
├─ README.md

We can rethink further once the python + yml, which will lilkely bring new considerations.